### ISSUE TITLE
-D does not delete last entry

### ISSUE THREAD

**Issue by solsticedhiver** [gene-git/nginx_passwd#3]
So I was trying to remove an entry with -D and that was the only one in the file, in fact.

The script output 'no data to print out', but the entry is not deleted from file.
```shell
$ cat htpasswd 
test:$5$rounds=535000$JpUfzpY4HgecbRoF$A5cAqGhVJ/JojnlQ/PYqZKBnDCbWbBgK3kclX4IryV6
solstice@rhea:/tmp$ nginx-passwd -f htpasswd -D test
no data to print
solstice@rhea:/tmp$ cat htpasswd 
test:$5$rounds=535000$JpUfzpY4HgecbRoF$A5cAqGhVJ/JojnlQ/PYqZKBnDCbWbBgK3kclX4IryV6
```

It's because `lib/utils.py` *write_passwd_file()* returns too soon if there is no data.

I would apply a patch like that, for example ?

```diff
diff --git a/src/nginx_passwd/lib/utils.py b/src/nginx_passwd/lib/utils.py
index 7341a60..76a5c0d 100644
--- a/src/nginx_passwd/lib/utils.py
+++ b/src/nginx_passwd/lib/utils.py
@@ -44,10 +44,6 @@ def write_passwd_file(passwd_data, file):
     """
     Write dictionary to file
     """
-    if not passwd_data:
-        print('no data to print')
-        return
-
     fobj =  open_file(file, 'w')
     if not fobj:
         print(f'Failed to open password file for writing : {file}')
@@ -59,6 +55,9 @@ def write_passwd_file(passwd_data, file):
         row = f'{user}:{passw}'
         fobj.write(row)
 
+    if not passwd_data:
+        print('no data to print')
+        return
 
 def print_passwd_file(passwd_data):
     """

```

**Comment by gene-git** [comment]
Good catch - and fix looks good too -  thank you - will incorporate and push shortly

**Comment by gene-git** [comment]
Fixed in release 2.5.0 - thank you again!
